src/librustdoc/html/static/main.js

```diff
@@ -90,6 +104,35 @@ if (!DOMTokenList.prototype.remove) {
         return null;
     }
 
+    function getCratesHash() {
+        if (window.hashCode === undefined || window.hashCode === 0) {
+            if (window.crates === undefined || window.crates.length === 0) {
+                window.crates = cloneArray(document.getElementById("crate-search")
+                                                   .getElementsByTagName("option"), 1)
```

Comment:
AFAICT, we wouldn't actually break anything if we didn't skip the first element in the options list here (it's always the same, sure, but that doesn't hurt). Can we do that to simplify the code here a little? It should also allow dropping the `skip` arg from `cloneArray` which is pretty irrelevant to the main purpose of the function.

Reply:
It adds a useless element into the hash computation and slows it down. If we remove the hash system, then this skip will become useless indeed but for the moment it's important to keep it. :)

Reply:
The cost of hashing this string is likely to be near-zero compared to all the other JS and such that we're running. I would rather pay that cost and have simpler to read code; as-is we at the very least should have a comment of "strips out default 'all crates' at the beginning.